packages/contracts/contracts/voting/GovernancePlugin.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
+
+import {Plugin} from "../core/plugin/Plugin.sol";
+import {IDAO} from "../core/IDAO.sol";
+import {IProposal} from "./IProposal.sol";
+import {GovernanceBase} from "./GovernanceBase.sol";
+
+abstract contract GovernancePlugin is GovernanceBase, Plugin {
+    using Counters for Counters.Counter;
+
+    /// @notice The incremental ID for proposals and executions.
+    Counters.Counter private proposalCounter;
+
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
+    bytes4 internal constant GOVERNANCE_INTERFACE_ID = type(IProposal).interfaceId;
+
+    /// @notice Checks if this or the parent contract supports an interface by its ID.
+    /// @param interfaceId The ID of the interace.
+    /// @return bool Returns true if the interface is supported.
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return interfaceId == GOVERNANCE_INTERFACE_ID || super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc IProposal
+    function proposalCount() external view returns (uint256) {
+        return proposalCounter.current();
+    }
+
+    /// @notice Constructs the plugin by storing the associated DAO.
+    /// @param _dao The DAO contract.
+    constructor(IDAO _dao) Plugin(_dao) {}
+
+    /// @inheritdoc GovernanceBase
+    function _createProposal(
```

Comment:
Only `_createProposal` could be moved since `_executeProposal` need the `dao` variable inherited from `Plugin`.

Moved in 6f81b9edcdcebe1791d918938d5c3c4fb1d16017